### PR TITLE
Bump package versions for release 2018-1-23

### PR DESCRIPTION
**Pull request by jasmineymlo**

### CODE REVIEW

File: provisioning/device/src/Microsoft.Azure.Devices.Provisioning.Client.csproj [hunk 1]
```diff
@@ -20,7 +20,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <VersionPrefix>1.0.1</VersionPrefix>
+    <VersionPrefix>1.0.2</VersionPrefix>
```
Comment (CIPop) [inline provisioning/device/src/Microsoft.Azure.Devices.Provisioning.Client.csproj]: Were there any ProvisioningClient versions? Do we need to release this?

File: security/tpm/src/Microsoft.Azure.Devices.Provisioning.Security.Tpm.csproj [hunk 2]
```diff
@@ -26,7 +26,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <VersionPrefix>1.0.1</VersionPrefix>
+    <VersionPrefix>1.0.2</VersionPrefix>
```
Comment (CIPop) [inline security/tpm/src/Microsoft.Azure.Devices.Provisioning.Security.Tpm.csproj]: Same questions as above - why do we need to release Transports and Security? Was there a fix?

File: provisioning/device/src/Microsoft.Azure.Devices.Provisioning.Client.csproj (hunk 1 above)
Comment (jasmineymlo) [reply provisioning/device/src/Microsoft.Azure.Devices.Provisioning.Client.csproj]: Removing unnecessary CodeAnalysis.FxCopAnalyzers IOperation declaration changed the csproj.  I didn't look at every change in details.  There were some changes in "shared" in other PR.  If you are sure they do not need to be updated/release, we could revert.

File: security/tpm/src/Microsoft.Azure.Devices.Provisioning.Security.Tpm.csproj (hunk 2 above)
Comment (jasmineymlo) [reply security/tpm/src/Microsoft.Azure.Devices.Provisioning.Security.Tpm.csproj]: same as above.

File: provisioning/device/src/Microsoft.Azure.Devices.Provisioning.Client.csproj (hunk 1 above)
Comment (CIPop) [reply provisioning/device/src/Microsoft.Azure.Devices.Provisioning.Client.csproj]: Discussed offline: OK to leave as is. The FXCop change should not affect the behavior but there were changes to shared that might have been functional.

**Pull request merged**